From fd2db4cd61cb96694de4d196fcb5d455b46bfa5f Mon Sep 17 00:00:00 2001
From: Zhou Peng <eagle.zhou@nxp.com>
Date: Thu, 26 Jul 2018 13:35:13 +0800
Subject: [PATCH 4242/5242] MLK-19058 - [I.MX8QXP/VPU]: Refine negotiation
 protocol to avoid camera reset before start

commit  428a1bd9cdc673f26dd999fd9f944d80c13d20d3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Set default colorspace value in v4l2_ioctl_try_fmt()

Signed-off-by: Zhou Peng <eagle.zhou@nxp.com>
(cherry picked from commit 52221438736a3ede393ba1682eadc292d3a73977)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index d968bb5..c7d3724 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -465,6 +465,7 @@ static int v4l2_ioctl_try_fmt(struct file *file,
 		struct v4l2_format *f
 		)
 {
+	struct v4l2_pix_format_mplane   *pix_mp = &f->fmt.pix_mp;
 	unsigned int table_size;
 
 	if (f->type == V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE) {
@@ -472,6 +473,8 @@ static int v4l2_ioctl_try_fmt(struct file *file,
 		if (!format_is_support(formats_compressed_enc, table_size, f))
 			return -EINVAL;
 	} else if (f->type == V4L2_BUF_TYPE_VIDEO_OUTPUT_MPLANE) {
+		pix_mp->field = V4L2_FIELD_ANY;
+		pix_mp->colorspace = V4L2_COLORSPACE_REC709;
 		table_size = ARRAY_SIZE(formats_yuv_enc);
 		if (!format_is_support(formats_yuv_enc, table_size, f))
 			return -EINVAL;
-- 
1.7.9.5

